vm/systemSmartContracts/delegation.go

```diff
@@ -377,6 +393,40 @@ func (d *delegation) delegateUser(
 	return vmcommon.Ok
 }
 
+func (d *delegation) makeStakeArgsIfAutomaticActivation(
+	config *DelegationConfig,
+	status *DelegationContractStatus,
+	globalFund *GlobalFundData,
+) [][]byte {
+	if !config.AutomaticActivation || len(status.NotStakedKeys) == 0 {
+		return nil
+	}
+
+	maxNodesToStake := big.NewInt(0).Div(globalFund.TotalActive, d.nodePrice).Uint64()
+	numStakedNodes := uint64(len(status.StakedKeys))
+	if maxNodesToStake <= numStakedNodes {
+		return nil
+	}
+
+	lenNotStakedKeys := uint64(len(status.NotStakedKeys)) + uint64(len(status.UnStakedKeys))
```

Comment:
why are we taking into account the unStakedKeys here?

Reply:
these are all the keys which are stakeable.

Reply:
done on line 401